examples/arduino-mega2560/src/bin/mega2560-rgb-led.rs

```diff
@@ -0,0 +1,67 @@
+#![no_std]
+#![no_main]
+
+use panic_halt as _;
+
+use arduino_hal::simple_pwm::IntoPwmPin;
+use arduino_hal::simple_pwm::Prescaler;
+use arduino_hal::simple_pwm::{Timer3Pwm, Timer4Pwm};
+
+#[arduino_hal::entry]
+fn main() -> ! {
+    let dp = arduino_hal::Peripherals::take().unwrap();
+    let pins = arduino_hal::pins!(dp);
+
+    let timer0 = Timer4Pwm::new(dp.TC4, Prescaler::Prescale64);
+    let timer1 = Timer3Pwm::new(dp.TC3, Prescaler::Prescale64);
+    let mut d6 = pins.d6.into_output().into_pwm(&timer0);
+    let mut d5 = pins.d5.into_output().into_pwm(&timer1);
+    let mut d3 = pins.d3.into_output().into_pwm(&timer1);
+
+    let max_duty_d6 = d6.get_max_duty();
+    let max_duty_d5 = d5.get_max_duty();
+    let max_duty_d3 = d3.get_max_duty();
+    let delay_time = 10;
+
+    d6.enable();
+    d5.enable();
+    d3.enable();
+
+    loop {
+        // Fade in red
+        for i in 0..=max_duty_d6 {
+            d6.set_duty(i);
+            arduino_hal::delay_ms(delay_time);
+        }
+
+        // Fade out red
+        for i in (0..=max_duty_d6).rev() {
+            d6.set_duty(i);
+            arduino_hal::delay_ms(delay_time);
+        }
+
+        // Fade in green
+        for i in 0..max_duty_d5 {
+            d5.set_duty(i);
+            arduino_hal::delay_ms(delay_time)
+        }
+
+        // Fade out green
+        for i in (0..max_duty_d5).rev() {
+            d5.set_duty(i);
+            arduino_hal::delay_ms(delay_time)
+        }
+
+        // Fade in blue
+        for i in 0..max_duty_d3 {
+            d3.set_duty(i);
+            arduino_hal::delay_ms(delay_time)
+        }
+
+        // Fade out blue
+        for i in (0..max_duty_d3).rev() {
+            d3.set_duty(i);
+            arduino_hal::delay_ms(delay_time)
+        }
```

Comment:
Why are you not using `..=` ranges for these loops as well?

Reply:
I've refactored this to better match the existing simple_pwm example for the uno